spinn_front_end_common/utility_models/lpg_splitter.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2020-2021 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from spinn_utilities.overrides import overrides
+from pacman.model.partitioner_splitters.abstract_splitters import (
+    AbstractSplitterCommon)
+from pacman.model.placements import Placement
+from pacman.utilities.algorithm_utilities.routing_algorithm_utilities import (
+    vertex_xy)
+from collections import defaultdict
+from .live_packet_gather_machine_vertex import LivePacketGatherMachineVertex
+
+
+class LPGSplitter(AbstractSplitterCommon):
```

Comment:
A short amount of documentation would be good here, but it could wait for a later PR if necessary.

Reply:
Fixed (and documented other functions too)